hyperspy/io_plugins/tvips.py

```diff
@@ -0,0 +1,347 @@
+    DPU = "1/m" if mode == 2 else "m"
+    SDP = header["pixelsize"][0]
+    offsetx = header["offsetx"][0]
+    offsety = header["offsety"][0]
+    # modify the data if there is scan information
+    # we construct a 2D array of indices to slice the data_stack
+    if scan_shape is not None:
+        # try to deduce start and stop of the scan based on rotator index
+        if scan_shape == "auto":
+            record_idxs = np.concatenate([i["rotidx"] for i in all_metadata])
+            scan_start_frame, scan_stop_frame = _find_auto_scan_start_stop(record_idxs)
+            if scan_start_frame is None or scan_stop_frame is None:
+                raise ValueError("Scan start and stop information could not be automatically "
+                        "determined. Please supply a scan_shape and scan_start_frame.")
+            total_scan_frames = record_idxs[scan_stop_frame]  # last rotator
+            scan_dim = int(np.sqrt(total_scan_frames))
+            if not np.allclose(scan_dim, np.sqrt(total_scan_frames)):
+                raise ValueError("Scan was not square, please supply a scan_shape and start_frame.")
+            scan_shape = (scan_dim, scan_dim)
+            # there may be discontinuities which must be filled up
+            indices = _guess_scan_index_grid(record_idxs, scan_start_frame, scan_stop_frame)
+        # scan shape and start are provided
+        else:
+            total_scan_frames = scan_shape[0] * scan_shape[1]
+            indices = np.arange(scan_start_frame, scan_start_frame+total_scan_frames).reshape(scan_shape[0], scan_shape[1])
+
+        # with winding scan, every second column or row must be inverted
+        # due to hysteresis there is also a predictable offset
+        if winding_scan_axis is not None:
+            if winding_scan_axis in ["x", 0]:
+                indices[::2] = indices[::2][:, ::-1]
+                indices[::2] = np.roll(indices[::2], hysteresis, axis=1)
+            elif winding_scan_axis in ["y", 1]:
+                indices[:, ::2] = indices[:, ::2][::-1, :]
+                indices[:, ::2] = np.roll(indices[:, ::2], hysteresis, axis=0)
+            else:
+                raise ValueError("Invalid winding scan axis")
+        
+        with dask.config.set(**{'array.slicing.split_large_chunks': True}):
+            data_stack = data_stack[indices.ravel()]
+        data_stack = data_stack.reshape(*indices.shape, dimy, dimx)
+        units = ['nm', 'nm', DPU, DPU]
+        names = ['y', 'x', 'dy', 'dx']
+        # no scale information stored in the scan!
+        scales = [1, 1, SDP, SDP]
+        offsets = [0, 0, offsety, offsetx]
+        # Create the axis objects for each axis
+        dim = data_stack.ndim
+        axes = [
+            {
+                'size': data_stack.shape[i],
+                'index_in_array': i,
+                'name': names[i],
+                'scale': scales[i],
+                'offset': offsets[i],
+                'units': units[i], }
```

Comment:
Following up on discussion in https://github.com/din14970/TVIPSconverter/issues/15#issuecomment-916041983, in these dictionary, a value should be defined for the `'navigate'` key to set the navigation axes (`'navigate'=True` means that this axis is a navigation axes!).

Reply:
This addition fixed things for me:
```'navigate': True if i < len(scan_shape) else False```